<?xml version="1.0"?>
<launch>

  <group ns="base_controller">
    <rosparam file="$(find robotino_controller_configuration_gazebo)/controller/base_controller.yaml" command="load" />
    <node name="joint_state_base_controller_spawner" pkg="controller_manager" type="spawner" args="joint_state_controller" respawn="true" />
    <node name="base_controller_vel_spawner" pkg="controller_manager" type="spawner" args="wheel0_joint_velocity_controller wheel1_joint_velocity_controller wheel2_joint_velocity_controller" respawn="true"/>
    <node name="joint_states_relay" pkg="topic_tools" type="relay" args="joint_states /joint_states" cwd="node" respawn="true" output="screen"/>   
    <node pkg="robotino_controller_configuration_gazebo" name="robotino_controller" type="base_sim_controller" output="screen">
      <remap from="wheel0" to="wheel0_joint_velocity_controller/command" />
      <remap from="wheel1" to="wheel1_joint_velocity_controller/command" />
      <remap from="wheel2" to="wheel2_joint_velocity_controller/command" />
    </node>
    <node pkg="robotino_controller_configuration_gazebo" name="odometry_controller" type="base_sim_odometry" output="screen">
      <remap from="odom" to="/odom" />
    </node>

  </group>

</launch>
